### PR TITLE
fix: avoid PRs when there's no contributions

### LINKED ISSUE #31: When a user is found, but the bot cannot determine contributions, tell the user
**Is your feature request related to a problem? Please describe.**
Sometimes when asking the bot to add contributions it won't be able to determine the types of contributions, this does not stop the bot opening a PR with an empty array of contributions see https://github.com/all-contributors/all-contributors-bot/pull/29/files#diff-c47fb5a6d4fde805d2706fc934e8d354R38

**Describe the solution you'd like**
When a user is found, but the bot cannot determine contributions, tell the user. A PR should not be opened.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Berkmann18**
This should resolve #31 (without the mess that #45 has).
## How?
By checking, right before submitting a PR and commenting, if the `contributions` array is empty or not.
If it is, it will comment accordingly otherwise proceed as usual.
I added a todo (not in the pushed commit tho) for [@sinchang 's case](https://github.com/all-contributors/all-contributors-bot/pull/45#issuecomment-454487100) as well (ie. contribution array unchanged).

### CODE REVIEW

File: src/tasks/processIssueComment/probot-processIssueComment.js [hunk 1]
```diff
@@ -24,6 +24,13 @@ async function processAddContributor({
     contributions,
     branchName,
 }) {
+    if (!contributions.length) {
+        context.log.debug('No contributions')
+        return commentReply.reply(
+            `I couldn't determine any contributions to add, did you specify any contributions?
+            Please make sure to use [valid contribution names](https://github.com/all-contributors/all-contributors#emoji-key-).`,
```
Comment (jakebolam) [inline src/tasks/processIssueComment/probot-processIssueComment.js]: Website is up! https://all-contributors.js.org/docs/emoji-key

Comment (jakebolam) [inline src/tasks/processIssueComment/probot-processIssueComment.js]: `contribution name`

File: src/tasks/processIssueComment/probot-processIssueComment.js [hunk 2]
```diff
@@ -24,6 +24,13 @@ async function processAddContributor({
     contributions,
     branchName,
 }) {
+    if (!contributions.length) {
```
Comment (jakebolam) [inline src/tasks/processIssueComment/probot-processIssueComment.js]: `if (!contributions.length > 0) {`

File: src/tasks/processIssueComment/probot-processIssueComment.js [hunk 3]
```diff
@@ -24,6 +24,13 @@ async function processAddContributor({
...
+        context.log.debug('No contributions')
+        return commentReply.reply(
+            `I couldn't determine any contributions to add, did you specify any contributions?
```
[the remark this answers is not in the document]
Comment (Berkmann18) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: Wouldn't it be a bit obscure (ie. black boxy) and not really English to use "did you specify contributions", considering people using the bot might not know that the array behind that part of the sentence is stored in a `contributions` variable (which would be a bit clearer if it was "did you specify `contributions`")?

File: src/tasks/processIssueComment/probot-processIssueComment.js (hunk 2 above)
Comment (Berkmann18) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: If it needs to be more explicit, than `!contributions.length`, wouldn't `contributions.length === 0` be crystal clear while having the same effect as `!contributions.length`?

Comment (jakebolam) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: Yep that would be clear too!

This is a habit from my C days where i'd never use == 0. Totally acceptable here!

File: src/tasks/processIssueComment/probot-processIssueComment.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakebolam) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: I think your right, leaving `any` is the right call

File: src/tasks/processIssueComment/probot-processIssueComment.js (hunk 2 above)
Comment (Berkmann18) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: Ah, I see!

**Comment by Berkmann18** [comment]
@jakebolam Do you know if #1 is useable to test this PR?
I've had a look at the `.json` fixtures and what the API returns, but it seems to be directed to you so I'm unsure if by commenting on it it would break it along with the other relevant test cases or if it's necessary to create another PR/issue thread with the comment(s) that would trigger that execution path.

File: src/tasks/processIssueComment/utils/parse-comment/index.js [hunk 4]
```diff
@@ -66,6 +66,7 @@ const contributionTypeMappings = {
     testing: 'test',
     tools: 'tool',
     tooling: 'tool',
+    translating: 'translation',
```
Comment (jakebolam) [inline src/tasks/processIssueComment/utils/parse-comment/index.js]: unsure why master changes are showing up in diffs in this PR, how are you pulling commits in from master?

Comment (jakebolam) [reply src/tasks/processIssueComment/utils/parse-comment/index.js]: @Berkmann18 
For me:
1. `git remote add parent git@github.com:all-contributors/all-contributors-bot.git`
2. `git fetch parent`
3. `git checkout master` or `git checkout empty-contribs`
4. `git merge parent/master`

See https://help.github.com/articles/syncing-a-fork/

File: src/tasks/processIssueComment/probot-processIssueComment.js [hunk 5]
```diff
@@ -24,6 +24,13 @@ async function processAddContributor({
     contributions,
     branchName,
 }) {
+    if (contributions.length === 0) {
```
Comment (jakebolam) [inline src/tasks/processIssueComment/probot-processIssueComment.js]: @Berkmann18 this code looks good to me. If we can fix the PR commit/merge issues then good to merge! :) 🎉

Comment (jakebolam) [reply src/tasks/processIssueComment/probot-processIssueComment.js]: Resolved, woo 🎉, go github

File: src/tasks/processIssueComment/utils/parse-comment/index.js (hunk 4 above)
Comment (Berkmann18) [reply src/tasks/processIssueComment/utils/parse-comment/index.js]: `git rebase master`

Comment (jakebolam) [reply src/tasks/processIssueComment/utils/parse-comment/index.js]: Perhaps needs to be `git rebase parent/master` on forks?

**Comment by jakebolam** [comment]
Working on sandbox staging 🎉 

screen shot 2019-01-30 at 9 55 53 am

**Comment by jakebolam** [comment]
Heading to production!
https://circleci.com/workflow-run/f00014c6-e047-471a-a209-209f8a6b85e0

File: src/tasks/processIssueComment/utils/parse-comment/index.js (hunk 4 above)
Comment (Berkmann18) [reply src/tasks/processIssueComment/utils/parse-comment/index.js]: Perhaps, I usually have the upstream set to `org`/`origin` pointing to `https://github.com/all-contributors/...` and usually don't get this problem.

**Pull request merged**